rclcpp/test/rclcpp/test_add_callback_groups_to_executor.cpp

```diff
@@ -204,18 +204,19 @@ TYPED_TEST(TestAddCallbackGroupsToExecutor, add_callback_groups_after_add_node_t
   auto node = std::make_shared<rclcpp::Node>("my_node", "/ns");
   executor.add_node(node->get_node_base_interface());
   ASSERT_EQ(executor.get_all_callback_groups().size(), 1u);
-  std::atomic_int timer_count {0};
+  std::atomic_size_t timer_count {0};
   auto timer_callback = [&executor, &timer_count]() {
+      printf("in timer_callback(%zu)\n", timer_count.load());
       if (timer_count > 0) {
-        ASSERT_EQ(executor.get_all_callback_groups().size(), 3u);
+        ASSERT_GT(executor.get_all_callback_groups().size(), 1u);
         executor.cancel();
       }
       timer_count++;
     };
   rclcpp::CallbackGroup::SharedPtr cb_grp = node->create_callback_group(
     rclcpp::CallbackGroupType::MutuallyExclusive);
   rclcpp::TimerBase::SharedPtr timer_ = node->create_wall_timer(
-    2s, timer_callback, cb_grp);
```

Comment:
Is this change needed?

Reply:
No this one is not necessary, though I believe it shortens the runtime of the test. Let me have another look at it. I'll try to avoid unnecessary changes.

But I might make another pr for this, and I may even shorten it further.

I think further changes to this test could also be made, since it really doesn't need the extra timers (adding more timers/cbgs after spin, which is the purpose of the test, isn't dependent on adding more than one, and frankly could hide an issue where only one added cbg does not work).